lhotse/dataset/fields.py

```diff
@@ -0,0 +1,338 @@
+    During collation, it will add `<PAD>` symbol ids at the end to match the longest
+    sequence in the batch.
+
+    Output:
+
+    .. code-block:
+
+        output = {'char_ids': List[int], 'char_ids_len': int}
+    """
+
+    def __init__(self, cuts):
+        self.pad = '<PAD>'
+        self.id2sym = sorted({char for cut in cuts for s in cut.supervisions for char in s.text}) + [self.pad]
+        self.sym2id = {v: k for k, v in enumerate(self.id2sym)}
+
+    def __call__(self, supervision: SupervisionSegment, cut: AnyCut) -> Dict[str, Any]:
+        return {'char_ids': [self.sym2id[c] for c in supervision.text], 'char_ids_len': len(supervision.text)}
+
+    def collate(self, items: List[int], key: str):
+        assert key in ('char_ids', 'char_ids_len')
+        if key == 'char_ids':
+            items = sorted(items, key=len, reverse=True)
+            for idx in range(1, len(items)):
+                items[idx].extend([self.sym2id[self.pad]] * (len(items[0]) - len(items[idx])))
+            items = torch.tensor(items, dtype=torch.int32)
+        elif key == 'chars_ids_len':
+            items = default_collate(items)
+        return items
+
+
+class Speaker:
+    """
+    Returns a field with the speaker ID index.
+    The indices are obtained by inspecting all speaker in the CutSet provided to ``__init__``.
+
+    Output:
+
+    .. code-block:
+
+        output = {'speaker': int}
+    """
+
+    def __init__(self, cuts: CutSet):
+        self.id2spk = sorted(cuts.speakers)
+        self.spk2id = {v: k for k, v in enumerate(self.id2spk)}
+
+    def __call__(self, supervision: SupervisionSegment, cut: AnyCut) -> Dict[str, Any]:
+        return {'speaker': self.spk2id[supervision.speaker]}
+
+
+class VoiceActivity:
+    """
+    Returns a field with a voice activity vector.
+    It is a binary vector that has values corresponding to each frame/sample.
+
+    Output:
+
+    .. code-block:
+
+        output = {'audio_is_voiced': <array, shape (T)>}
+        output = {'frame_is_voiced': <array, shape (T)>}
+    """
+
+    def __init__(self, use_features: bool = True, use_audio: bool = False):
+        if not use_audio and not use_features:
+            raise ValueError("VoiceActivity requested but both use_audio and use_features set to False.")
+        self.use_features = use_features
+        self.use_audio = use_audio
+
+    def __call__(self, supervision: SupervisionSegment, cut: AnyCut) -> Dict[str, Any]:
+        output = {}
+        if self.use_audio:
+            output['audio_is_voiced'] = cut.supervisions_audio_mask()
+        if self.use_features:
+            output['frame_is_voiced'] = cut.supervisions_feature_mask()
+        return output
+
+
+def _collate_features(cuts: CutSet) -> torch.Tensor:
+    assert all(cut.has_features for cut in cuts)
+    first_cut = next(iter(cuts))
+    features = torch.empty(len(cuts), first_cut.num_frames, first_cut.num_features)
```

Comment:
Perhaps a comment could clarify this?  Are we assuming the cuts all have the same length, or that the 1st one is longest [if the latter, the use of `torch.empty()` is problematic]